pdc/apps/auth/views.py

```diff
@@ -73,8 +73,41 @@ def logout(request):
     return HttpResponseRedirect(redirect_to)
 
 
+def _get_resource_permissions_matrix(user):
+    http_method = ['create', 'read', 'update', 'delete']
+    resource_permission_set = set([])
+    dict = {}
+    result = []
+
+    if user.is_superuser:
+        resource_permission_set = set([obj for obj in models.ResourcePermission.objects.all()])
+    else:
+        group_id_list = [group.id for group in user.groups.all()]
+        queryset = models.GroupResourcePermission.objects.filter(group__id__in=group_id_list)
+        for group_resource_permission in queryset:
+            resource_permission_set.add(group_resource_permission.resource_permission)
```

Comment:
Above part from `if`, looks almost seem as the logic in `current-user` API. Could we extract the logic to a method and reuse it in both places to eliminate duplicate code. Refactor our code a little.